Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -1107,16 +1296,34 @@ private void AddLinks()
                     ball.name = "midpoint_" + i.ToString();
                     ball.transform.localPosition = edgeCenters[i];
 
+                    // Align handle with its edge assuming that the prefab is initially aligned with the up direction 
+                    if (edgeAxes[i] == CardinalAxisType.X)
+                    {
+                        Quaternion realignment = Quaternion.FromToRotation(Vector3.up, Vector3.right);
+                        ball.transform.localRotation = realignment * ball.transform.localRotation;
+                    }
+                    else if (edgeAxes[i] == CardinalAxisType.Z)
+                    {
+                        Quaternion realignment = Quaternion.FromToRotation(Vector3.up, Vector3.forward);
+                        ball.transform.localRotation = realignment * ball.transform.localRotation;
+                    }
+
                     if (rotationHandlePrefabColliderType == RotationHandlePrefabCollider.Sphere)
                     {
-                        SphereCollider collider = ball.AddComponent<SphereCollider>();
-                        collider.radius = 0.5f * rotationHandleDiameter;
+                        SphereCollider sphereCollider = ball.AddComponent<SphereCollider>();
+                        sphereCollider.radius = 0.5f * rotationHandleDiameter;
+                        float localScale = ball.transform.localScale.x;
+                        localScale = localScale == 0.0f ? 0.0001f : localScale;
```

Comment:
What is this for? Can you explain via  a comment? I am not immediately clear why if localScale is zero we force it to be non-zero

Reply:
when localScale was zero there was a divide error that caused code to be skipped over when the exception threw. I guess this current code could be replaced with :
localScale = Mathf.Max(0.0001f, localScale);